src/Jobs/OptimizeHtml.php

```diff
@@ -1,25 +1,37 @@
 <?php
 
-namespace SiteOrigin\PageCache\Listeners;
+namespace SiteOrigin\PageCache\Jobs;
 
+use Illuminate\Bus\Queueable;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
 use Illuminate\Support\Collection;
 use SiteOrigin\PageCache\Events\PageOptimized;
 use SiteOrigin\PageCache\Events\PageOptimizing;
-use SiteOrigin\PageCache\Events\PageRefreshed;
 use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUniqueUntilProcessing;
+use SiteOrigin\PageCache\Page;
 
-class OptimizeHtml implements ShouldQueue
+class OptimizeHtml implements ShouldQueue, ShouldBeUniqueUntilProcessing
 {
-    public function handle(PageRefreshed $event)
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected Page $page;
+
+    public function __construct(Page $page)
     {
-        $page = $event->getPage();
+        $this->page = $page;
+    }
 
-        if ($page->getFileExtension() == 'html' && $this->hasOptimizers()) {
+    public function handle()
+    {
+        if ($this->page->getFileExtension() == 'html' && $this->hasOptimizers()) {
             // Create a temporary file with the original filename
             $filename = tempnam('', 'page-cache');
```

Comment:
I have been thinking about this, and maybe we should do it now, but maybe it would be better to use [tmpfile](https://www.php.net/manual/en/function.tmpfile.php) for our temporary file instead? This means each optimizer would need to deal with a file handle instead.

Optimizers that need a filename, would just use [stream_get_meta_data](https://www.php.net/manual/en/function.stream-get-meta-data.php) like this: https://stackoverflow.com/a/7690726

And then just closing the file handle at the end will automatically remove it from the filesystem. It all just seems safer.

Reply:
Actually, I'll log this for a later change.